src/Storage/EntryModel.php

```diff
@@ -1,182 +1,7 @@
-            return $query->where('type', $type);
-        });
-
-        return $this;
-    }
-
-    /**
-     * Scope the query for the given batch ID.
-     *
-     * @param  \Illuminate\Database\Eloquent\Builder  $query
-     * @param  \Laravel\Telescope\Storage\EntryQueryOptions  $options
-     * @return $this
-     */
-    protected function whereBatchId($query, EntryQueryOptions $options)
-    {
-        $query->when($options->batchId, function ($query, $batchId) {
-            return $query->where('batch_id', $batchId);
-        });
-
-        return $this;
-    }
-
-    /**
-     * Scope the query for the given type.
-     *
-     * @param  \Illuminate\Database\Eloquent\Builder  $query
-     * @param  \Laravel\Telescope\Storage\EntryQueryOptions  $options
-     * @return $this
-     */
-    protected function whereTag($query, EntryQueryOptions $options)
-    {
-        $query->when($options->tag, function ($query, $tag) {
-            return $query->whereIn('uuid', function ($query) use ($tag) {
-                $query->select('entry_uuid')->from('telescope_entries_tags')->whereTag($tag);
-            });
-        });
-
-        return $this;
-    }
-
-    /**
-     * Scope the query for the given type.
-     *
-     * @param  \Illuminate\Database\Eloquent\Builder  $query
-     * @param  \Laravel\Telescope\Storage\EntryQueryOptions  $options
-     * @return $this
-     */
-    protected function whereFamilyHash($query, EntryQueryOptions $options)
-    {
-        $query->when($options->familyHash, function ($query, $hash) {
-            return $query->where('family_hash', $hash);
-        });
-
-        return $this;
-    }
-
-    /**
-     * Scope the query for the given pagination options.
-     *
-     * @param  \Illuminate\Database\Eloquent\Builder  $query
-     * @param  \Laravel\Telescope\Storage\EntryQueryOptions  $options
-     * @return $this
-     */
-    protected function whereBeforeSequence($query, EntryQueryOptions $options)
-    {
-        $query->when($options->beforeSequence, function ($query, $beforeSequence) {
-            return $query->where('sequence', '<', $beforeSequence);
-        });
-
-        return $this;
-    }
-
-    /**
-     * Scope the query for the given display options.
-     *
-     * @param  \Illuminate\Database\Eloquent\Builder  $query
-     * @param  \Laravel\Telescope\Storage\EntryQueryOptions  $options
-     * @return $this
-     */
-    protected function filter($query, EntryQueryOptions $options)
-    {
-        if ($options->familyHash || $options->tag || $options->batchId) {
-            return $this;
-        }
-
-        return $this;
-    }
-
-    /**
-     * Get the current connection name for the model.
-     *
-     * @return string
-     */
-    public function getConnectionName()
-    {
-        return config('telescope.storage.database.connection');
-    }
+if (env('DB_CONNECTION') == 'mongodb') {
+    require_once 'MongoModel.php';
+} else {
+    require_once 'RegularModel.php';
 }
```

Comment:
Al posto di questo potresti fare una sorta di factory, del tipo:
```
public static function getModelClass()
{
    return env('DB_CONNECTION') === "mongodb" ? MongoModel::class : RegularModel::class;
}
```

Poi, dove attualmente viene utilizzato `EntryModel`, sostituisci di conseguenza. Ad esempio, in `DatabaseEntriesRepository:63` (e analogamente a riga 91) avresti:
```
$entry = EntryModel::getModelClass()::on($this->connection)->whereUuid($id)->firstOrFail();
```

Non so se ci ho preso, dimmi se ti torna!